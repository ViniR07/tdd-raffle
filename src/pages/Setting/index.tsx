import React from 'react';
import Card from '../../components/Card';
import Footer from '../../components/Footer';
import Form from '../../components/Form';
import ParticipantsList from '../../components/ParticipantsList';

// import { Container } from './styles';

const Setting: React.FC = () => {
  return (
    <Card>
      <section>
        <h2>Vamos come√ßar!</h2>
        <Form />
        <ParticipantsList />
        <Footer />
      </section>
    </Card>
  );
};

export default Setting;
