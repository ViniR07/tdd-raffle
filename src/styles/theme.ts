import {
	blackIdentity,
	blackTextColor,
	grayIdentity,
	orangeIdentity,
	purpleIdentity,
	whiteIdentity,
} from './colors';

const theme = {
	colors: {
		purple: purpleIdentity,
		text: blackTextColor,
		orange: orangeIdentity,
		black: blackIdentity,
		gray: grayIdentity,
		white: whiteIdentity,
	},
	fonts: {
		title: 'Poppins, sans serif',
		text: 'Poppins, sans serif',
		size: {
			sm: '16px',
			sm2: '20px',
			md: '25px',
			lg: '49px',
		},
	},
	// gradients: {
	// 	title:
	// 		'linear-gradient(121.57deg, #FFFFFF 18.77%, rgba(255, 255, 255, 0.66) 60.15%)',
	// 	primary: 'linear-gradient(to right, #8e2de2, #4a00e0)',
	// 	secondary1: 'linear-gradient(to right, #0052d4, #4364f7, #6fb1fc)',
	// },
	// Breakpoints for responsive design
	breakpoints: {
		sm: 'screen and (max-width: 640px)',
		md: 'screen and (max-width: 768px)',
		lg: 'screen and (max-width: 1024px)',
		xl: 'screen and (max-width: 1280px)',
	},
};

export default theme;
