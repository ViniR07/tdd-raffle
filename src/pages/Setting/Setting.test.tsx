import { render } from '@testing-library/react';
import React from 'react';
import { RecoilRoot } from 'recoil';
import { ThemeProvider } from 'styled-components';
import Setting from '.';
import theme from '../../styles/theme';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockNavigate,
  };
});

describe('in Setting page', () => {
  test('must be rendered correctly', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <RecoilRoot>
          <Setting />
        </RecoilRoot>
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
