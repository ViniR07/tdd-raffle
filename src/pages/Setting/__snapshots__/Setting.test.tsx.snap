// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`in Setting page must be rendered correctly 1`] = `
<div>
  <div
    class="sc-bczRLJ AhOXQ"
  >
    <section>
      <h2>
        Vamos come√ßar!
      </h2>
      <form>
        <div
          class="sc-gsnTZi einVYy"
        >
          <input
            placeholder="Insira os nomes dos participantes"
            type="text"
            value=""
          />
          <button
            disabled=""
            type="submit"
          >
            Adicionar
          </button>
        </div>
        
      </form>
      <ul />
      <footer
        class="sc-hKMtZM hvkwwq"
      >
        <button
          class="sc-dkzDqf kLILcq"
          disabled=""
        >
          Iniciar brincadeira!
        </button>
        <img
          alt="Sacolas de compras"
          src="/images/sacolas.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
